doc/SdA.txt

```diff
@@ -41,37 +41,40 @@ before going any further.
 Stacked Autoencoders
 ++++++++++++++++++++
 
-The denoising autoencoders can be stacked to form a deep network by
+Denoising autoencoders can be stacked to form a deep network by
 feeding the latent representation (output code)
-of the denoising auto-encoder found on the layer 
+of the denoising autoencoder found on the layer 
 below as input to the current layer. The **unsupervised pre-training** of such an 
 architecture is done one layer at a time. Each layer is trained as 
-a denoising auto-encoder by minimizing the reconstruction of its input
+a denoising autoencoder by minimizing the error in reconstructing its input
 (which is the output code of the previous layer).
 Once the first :math:`k` layers 
 are trained, we can train the :math:`k+1`-th layer because we can now 
 compute the code or latent representation from the layer below. 
+
 Once all layers are pre-trained, the network goes through a second stage
-of training called **fine-tuning**. Here we consider **supervised fine-tuning**
+of training called **fine-tuning**,
 where we want to minimize prediction error on a supervised task.
-For this we first add a logistic regression 
+For this, we first add a logistic regression 
 layer on top of the network (more precisely on the output code of the
 output layer). We then
 train the entire network as we would train a multilayer 
 perceptron. At this point, we only consider the encoding parts of
 each auto-encoder.
 This stage is supervised, since now we use the target class during
-training (see the :ref:`mlp` for details on the multilayer perceptron).
+training. (See the :ref:`mlp` for details on the multilayer perceptron.)
 
 This can be easily implemented in Theano, using the class defined
-before for a denoising autoencoder. We can see the stacked denoising
-autoencoder as having two facades, one is a list of
-autoencoders, the other is an MLP. During pre-training we use the first facade, i.e we treat our model
+previously for a denoising autoencoder. We can see the stacked denoising
+autoencoder as having two facades: One is a list of
+autoencoders. The other is an MLP. During pre-training we use the first facade, i.e., we treat our model
 as a list of autoencoders, and train each autoencoder seperately. In the 
 second stage of training, we use the second facade. These two
-facedes are linked by the fact that the autoencoders and the sigmoid layers of 
-the MLP share parameters, and the fact that autoencoders get as input latent
-representations of intermediate layers of the MLP. 
+facades are linked 
+
+* by the parameters shared by the autoencoders and the sigmoid layers of the MLP, and 
+
+* by feeding the latent representations of intermediate layers of the MLP as input to the autoencoders.
```

Comment:
They're not really "linked by the parameters", or "linked by feeding ...".
I would rather say something like:
    These two facades are linked because:
    * the autoencoders and the sigmoid layers of the MLP share parameters, and
    * the latent representations computed by intermediate layers of the MLP are fed as input to the autoencoders.
```